### PR TITLE
Issue #1445 - Cleanup on MvcOptions.ApplicationModelConventions.

### PR DESCRIPTION
**Pull request by sornaks**
#1445

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Core/ControllerActionDescriptorProvider.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ public class ControllerActionDescriptorProvider : IActionDescriptorProvider
             _assemblyProvider = assemblyProvider;
             _applicationModelBuilder = applicationModelBuilder;
             _globalFilters = globalFilters.Filters;
-            _modelConventions = optionsAccessor.Options.ApplicationModelConventions;
+            _modelConventions = optionsAccessor.Options.Conventions;
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/ControllerActionDescriptorProvider.cs]: rename the field too

File: src/Microsoft.AspNet.Mvc.Core/DefaultActionModelConvention.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Mvc.ApplicationModels;
+
+namespace Microsoft.AspNet.Mvc
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/DefaultActionModelConvention.cs]: put this in `Microsoft.AspNet.Mvc.ApplicationModels`

File: src/Microsoft.AspNet.Mvc.Core/DefaultControllerModelConvention.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Mvc.ApplicationModels;
+
+namespace Microsoft.AspNet.Mvc
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/DefaultControllerModelConvention.cs]: `Microsoft.AspNet.Mvc.ApplicationModels`

File: src/Microsoft.AspNet.Mvc.Core/DefaultControllerModelConvention.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+{
+    /// <summary>
+    /// The class used by <see cref="ApplicationModelConventionExtensions"/> to add a
+    /// <see cref="IControllerModelConvention"/> to all the controllers in the application.
+    /// </summary>
+    public class DefaultControllerModelConvention : IApplicationModelConvention
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/DefaultControllerModelConvention.cs]: I don't really love the name here, but I don't have a better idea. Maybe `ControllerApplicationModelConvention`? It's pretty bad too.

File: test/Microsoft.AspNet.Mvc.Core.Test/ApplicationModel/DefaultActionModelConventionTest.cs [hunk 5]
```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.AspNet.Mvc.ApplicationModels;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Core.Test/ApplicationModel/DefaultActionModelConventionTest.cs]: `Microsoft.AspNet.Mvc.ApplicationModels`

File: test/Microsoft.AspNet.Mvc.Core.Test/ApplicationModel/DefaultControllerModelConventionTest.cs [hunk 6]
```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.AspNet.Mvc.ApplicationModels;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Core.Test/ApplicationModel/DefaultControllerModelConventionTest.cs]: `Microsoft.AspNet.Mvc.ApplicationModels`

File: src/Microsoft.AspNet.Mvc.WebApiCompatShim/WebApiCompatShimOptionsSetup.cs [hunk 7]
```diff
@@ -22,10 +22,10 @@ public int Order
...
         {
             // Add webapi behaviors to controllers with the appropriate attributes
-            options.ApplicationModelConventions.Add(new WebApiActionConventionsApplicationModelConvention());
-            options.ApplicationModelConventions.Add(new WebApiParameterConventionsApplicationModelConvention());
-            options.ApplicationModelConventions.Add(new WebApiOverloadingApplicationModelConvention());
-            options.ApplicationModelConventions.Add(new WebApiRoutesApplicationModelConvention(area: DefaultAreaName));
+            options.Conventions.Add(new WebApiActionConventionsApplicationModelConvention());
+            options.Conventions.Add(new WebApiParameterConventionsApplicationModelConvention());
+            options.Conventions.Add(new WebApiOverloadingApplicationModelConvention());
+            options.Conventions.Add(new WebApiRoutesApplicationModelConvention(area: DefaultAreaName));
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.WebApiCompatShim/WebApiCompatShimOptionsSetup.cs]: if you want bonus points you can simplify these :grinning:

**Comment by rynowak** [comment]
:watch:

File: src/Microsoft.AspNet.Mvc.Core/DefaultActionModelConvention.cs (hunk 2 above)
Comment (yishaigalatzer) [inline src/Microsoft.AspNet.Mvc.Core/DefaultActionModelConvention.cs]: make this and the the DefaultControllerModelConvention private classes (and name them appropriately)

**Comment by sornaks** [comment]
Updated.. Please have a look.

File: test/Microsoft.AspNet.Mvc.Core.Test/ApplicationModel/ActionApplicationModelConventionTest.cs [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    public class ActionApplicationModelConventionTest
+    {
+        [Fact]
+        public void DefaultActionModelConvention_AppliesToAllActionsInApp()
+        {
+            // Arrange
+            var app = new ApplicationModel();
+            app.Controllers.Add(new ControllerModel(typeof(HelloController).GetTypeInfo(), new List<object>()));
+            app.Controllers.Add(new ControllerModel(typeof(WorldController).GetTypeInfo(), new List<object>()));
+            var defaultConvention = new ApplicationModelConventionExtensions.ActionApplicationModelConvention(
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Core.Test/ApplicationModel/ActionApplicationModelConventionTest.cs]: IMO, just think about using the extension method here and keep the class private. You don't have to try to hard to test this without making the classes internal

**Comment by rynowak** [comment]
:shipit: from me. I also have a suggestion about how to keep the implementation classes private which is worth considering

**Comment by sornaks** [comment]
Modified that and checked in - https://github.com/aspnet/Mvc/commit/d2c3985cd60d01e355c89ed7eb9d2f6922da762a

**Pull request closed without merging**